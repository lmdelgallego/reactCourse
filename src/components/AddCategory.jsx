import PropsTypes from 'prop-types';
import { useState } from "react"

function AddCategory({ onNewCategory }) {

  const [inputValue, setInputValue] = useState('One Punch')

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length < 2) return;
    onNewCategory( inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={ onSubmit } aria-label='form'>
      <input
        type="text"
        placeholder="Search..."
        value={inputValue}
        onChange={ onInputChange }
      />
    </form>
  )
}

export default AddCategory

AddCategory.propTypes = {
  onNewCategory: PropsTypes.func.isRequired
}