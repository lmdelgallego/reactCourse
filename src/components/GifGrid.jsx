import React from 'react'

const GifGrid = ({ category}) => {
  return (
    <>
      <div key={category}>
        <h3>{category}</h3>
        <li>{category}</li>
      </div>
    </>
  )
}

export default GifGrid