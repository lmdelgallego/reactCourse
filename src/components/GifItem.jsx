import PropsTypes from 'prop-types';

export const GifItem = ({ id, title, url  }) => {
  return (
    <div className='card' >
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropsTypes.string.isRequired,
  url: PropsTypes.string.isRequired
}