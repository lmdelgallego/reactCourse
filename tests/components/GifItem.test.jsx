import { render } from '@testing-library/react';
import {GifItem} from '../../src/components/GifItem';
describe('GifItem', () => {

  const url = 'https://localhost/algo.jpg';
  const title = 'Un titulo';

  test('should be match with snapshot', () => {

    const { container } = render(<GifItem title={ title } url={url} />)
    expect(container).toMatchSnapshot();
  });
});