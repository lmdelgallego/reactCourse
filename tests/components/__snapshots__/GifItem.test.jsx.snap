// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
